plugins/authentication/src/pam_password.cpp

```diff
@@ -287,8 +287,18 @@ namespace irods
 
             json resp{req};
 
+            // This operation asserts that the key exists in the request above, so we know that it is there.
+            const auto password_iter = resp.find(irods::AUTH_PASSWORD_KEY);
+            const auto password = password_iter->get<std::string>();
+
+            // Erase the password key as soon as is is no longer needed to avoid sending it back in the response.
+            resp.erase(password_iter);
+
             log_auth::trace("getting TTL param");
 
+            // Check TTL parameters before checking the password so that we avoid unnecessary communication with the PAM
+            // service. If we cannot authenticate with the iRODS server because of a bad TTL input, there was no reason
```

Comment:
not sure... the tense is correct?

```suggestion
            // service. If we cannot authenticate with the iRODS server because of a bad TTL input, there is no reason
```

Reply:
Done